snuba/datasets/storage.py

```diff
@@ -34,6 +37,8 @@ def get_schemas(self) -> StorageSchemas:
         """
         raise NotImplementedError
 
+    def get_cluster(self) -> Cluster:
+        return get_cluster(self.__storage_key)
```

Comment:
I think this would be confusing. It would make people think that storages are directly assigned to clusters while they are assigned only through StorageSet. I think you could either document that relationship here (in details) considering this is the method in the abstract class, or just return the storage set that just makes it clear.